tools/engine_tool/lib/src/build_utils.dart

```diff
@@ -203,12 +217,22 @@ Future<int> runGn(
     runTests: false,
   );
 
-  final bool buildResult = await buildRunner.run((RunnerEvent event) {
+  final bool built = await buildRunner.run((RunnerEvent event) {
     switch (event) {
       case RunnerResult(ok: false):
         environment.logger.error(event);
       default:
     }
   });
-  return buildResult ? 0 : 1;
+  if (built) {
+    if (!buildDir.existsSync()) {
+      environment.logger.error(
+        'The specified build did not produce the expected output directory: '
+        '${buildDir.path}',
+      );
+      return false;
+    }
+    return true;
+  }
+  return false;
```

Comment:
Small simplification.

```suggestion
  if (built && !buildDir.existsSync()) {
      environment.logger.error(
        'The specified build did not produce the expected output directory: '
        '${buildDir.path}',
      );
      return false;
  }
  return built;
```

Reply:
Yeah that's much better, done.